File: WrongException.cpp
// open a file and export the lines in fixed chunk sizes

#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main(int argc, char* argv[]) {
    string filename;
    int filesize, chunksize;
    ifstream file;
    cout << "Enter filename: ";
    cin >> filename;
    cout << "Enter file size: ";
    cin >> filesize;
    cout << "Enter read chunk size: ";
    cin >> chunksize;
    try {
        char buffer(chunksize);
        file.open(filename);
        if (!file.is_open()) 
            throw (string) "File not found";
        // determine the number of chunks needed to completely read the file
        // likely error: user enters zero for a chunksize creates a a floating point exception
        int numIters = (filesize/chunksize)+1;
        for (int i = 0; i <numIters; i++) {
            file.read(buffer,chunksize);
            printf("%s",buffer);
        }
    } catch (...) {
        cout << "file not found" << endl;
        exit(-1);
    }
}

File: WrongException.java
import java.io.*;
import java.lang.*;

// read a file in chunks and then print to command line
public class WrongException {
    public static void main(String args[]) {
        String filename = args[0];
        // allow a user input to determine the size of the buffer
        int chunkSize = Integer.parseInt (args[1]);
        // determines how many time the file needs to be read
        double iterations = 0.0;
        try {
            // file reader
            FileReader fr = new FileReader(filename);
            char a[] = new char[chunkSize];
            long fileLength = new File(filename).length();
            // calculate the length of the file
            // math.ceil rounds any value up to next whole number to complete necessary reads
            iterations = Math.ceil(fileLength/chunkSize);
            // actual functionality
            for (int i = 0; i < iterations; i++) {
                fr.read(a,0,chunkSize);
                System.out.print(a);
            }
            System.out.println();
        }
        catch (Exception e) {
            // in the event the file doesn't exist
            System.out.println ("File not found.");
        }
    }    
}

/*
 * calling the program: java WrongException <buffer size>
 * problem number 1: user provides a chunksize of zero creates an undefined chunkSize, catch block will still say the file is not found
 */
